### PR TITLE
Refactor GoogleMaps adapter to use GoogleMap directly for rendering GeoJson geometries.

### PR DESCRIPTION
**Pull request by nmhung00**
To bypass android utils library geo json layer implementation that has performance problem with lots of layers in the map while having strict mode on for API level 27+.

This PR should address the ANR on app open for projects with a lot of features. I didn't include the api level upgrade because there seems to be some other problem with map the OnCameraMoveStartedListener that causes some slowness when moving the map around.

No test was added since this is supposed to be a refactoring and no new functionality was added.

#946

### CODE REVIEW

**Comment by gino-m** [comment]
@shobhitagarwal1612 Is currently planning a refactor of the MapAdapter and related classes and has been looking at this most recently. Could you PTAL?

**Comment by shobhitagarwal1612** [comment]
@gino-m https://github.com/google/ground-android/pull/1005 doesn't touch the GoogleMapsMapAdapter. So this PR is non-conflicting.

**Review by scolsen — APPROVED** [review]
Nice!

File: gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java [hunk 1]
```diff
@@ -552,4 +484,108 @@ private void addRemoteTileOverlay(String url) {
   public void addRemoteTileOverlays(ImmutableList<String> urls) {
     stream(urls).forEach(this::addRemoteTileOverlay);
   }
+
+  /**
+   * A collection of geometries in a GeoJson feature.
+   */
+  private class GeoJsonFeatureGeometryCollection {
```
[the remark this answers is not in the document]
Comment (nmhung00) [reply gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java:491]: Yeah I can rename this class to `GeometryCollection`. The current name is too verbose and has unnecessary information.

[the remark this answers is not in the document]
Comment (nmhung00) [reply gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java]: I just pushed some changes to rename that class. Not quite sure if I'm following up the pull request workflow quite correctly so let me know if I should do some other things differently.

**Review by gino-m — APPROVED** [review]
Thanks for refactor!

**Pull request merged**